pixi.toml

```diff
@@ -23,20 +28,45 @@ rattler-build-conda-compat = ">=1.2.0,<2.0.0a0"
 [feature.linux]
 # The linux feature runs on every platform that can run Docker.
 platforms = ["linux-64", "linux-aarch64", "osx-64", "osx-arm64", "win-64"]
-[feature.linux.tasks]
-build-linux = { cmd = "python build-locally.py", env = { BUILD_LOCALLY_FILTER = "linux*" } }
+[feature.linux.tasks.build-linux]
+description = "build for Linux inside a docker container"
+cmd = "python build-locally.py"
+env = { BUILD_LOCALLY_FILTER = "linux*" }
 
 # The osx env will run conda-build directly on the machine.
 [feature.osx]
 platforms = ["osx-64", "osx-arm64"]
-[feature.osx.tasks]
-build-osx = { cmd = "python build-locally.py", env = { BUILD_LOCALLY_FILTER = "osx*", OSX_SDK_DIR = "$PIXI_PROJECT_ROOT/.pixi/macOS-SDKs", MINIFORGE_HOME = "$CONDA_PREFIX", CONDA_BLD_PATH = "$PIXI_PROJECT_ROOT/build_artifacts" } }
+[feature.osx.tasks.build-osx]
+description = "build directly on a MacOS host machine"
+cmd = "python build-locally.py"
+[feature.osx.tasks.build-osx.env]
+BUILD_LOCALLY_FILTER = "osx*"
+CONDA_BLD_PATH = "$PIXI_PROJECT_ROOT/build_artifacts"
+MINIFORGE_HOME = "$CONDA_PREFIX"
+OSX_SDK_DIR = "$PIXI_PROJECT_ROOT/.pixi/macOS-SDKs"
 
 # The windows env will run conda-build directly on the machine.
 [feature.win]
-platforms = ["win-64"] # , "win-arm64"]
-[feature.win.tasks]
-build-win = { cmd = "python build-locally.py", env = { BUILD_LOCALLY_FILTER = "win*", MINIFORGE_HOME = "$CONDA_PREFIX", CONDA_BLD_PATH = "$PIXI_PROJECT_ROOT/build_artifacts" } }
+platforms = ["win-64"]
+[feature.win.tasks.build-win]
+description = "build directly on a Windows host machine"
+cmd = "python build-locally.py"
+[feature.win.tasks.build-win.env]
+BUILD_LOCALLY_FILTER = "win*"
+CONDA_BLD_PATH = "$PIXI_PROJECT_ROOT/build_artifacts"
+MINIFORGE_HOME = "$CONDA_PREFIX"
+
+[feature.grayskull.dependencies]
+grayskull = ">=2.7.3"
+[feature.grayskull.tasks.grayskull]
+description = "build a recipe from a PyPI or CRAN package name with `grayskull`"
+cmd = "cd recipes && grayskull"
+
+[feature.lint.dependencies]
+conda-smithy = ">=3.44.6,<4"
+[feature.lint.tasks.recipe-lint]
+description = "validate all recipes with `conda-smithy`"
+cmd = "conda-smithy recipe-lint --conda-forge recipes/*"
```

Comment:
Staged-recipes has extra linting that is not captured in conda-smithy, see https://github.com/conda-forge/staged-recipes/blob/main/.github/workflows/scripts/linter.py. I think some parts of the linter there expect it to operate on an existing PR though, so it's not super straight-forward to add. We might need a little refactor in the script to ignore the PR parts unless the number is provided like it happens on CI.

Reply:
sure, but even stock `conda-smithy` will already find a lot of issues, e.g. with recent changes vs a `grayskull` output.

Reply:
Fair enough. For consistency with [conda-smithy](https://github.com/conda-forge/conda-smithy/blob/4c4402b304c4539599eeeaf15c3fbb31804194fe/conda_smithy/templates/pixi.toml.tmpl#L41-L43), let's call this task `lint` though.

Reply:
sure, task is now `lint`, feature is `conda-smithy`.